.devcontainer/python/Dockerfile

```diff
@@ -34,27 +33,21 @@ USER $USER
 ENV HOME="/home/$USER"
 ENV PATH="${HOME}/.local/bin:$PATH"
 
-# Set poetry related env vars
-# Set poetry virtualenv path in writable dir outsde mounted workspace
-ENV POETRY_VIRTUALENVS_PATH="$HOME/.venv"
-ENV POETRY_VIRTUALENVS_CREATE=true
-ENV POETRY_VIRTUALENVS_IN_PROJECT=false
-ENV POETRY_VIRTUALENVS_PROMPT='ecalc-py{python_version}'
 ENV PIP_DEFAULT_TIMEOUT=100
 
 # Python and poetry installation
 RUN echo $(which python) && echo $(which python3)  \
     && python3 -m pip install --upgrade pip pipx \
-    && pipx install "poetry==${POETRY_VERSION}" \
+    && pipx install "uv==${UV_VERSION}" \
```

Comment:
I'd recommend a different installation method in a Dockerfile: https://docs.astral.sh/uv/guides/integration/docker/#installing-uv

No need to install & update pip or pipx, and pre-commit can be installed via uv.